package/gluon-core/luasrc/usr/lib/lua/gluon/util.lua

```diff
@@ -188,4 +190,73 @@ function M.log(message, verbose)
 	posix_syslog.syslog(posix_syslog.LOG_INFO, message)
 end
 
+M.PipePolicies = {
+	DISCARD=-1,
+	INHERIT=0,
+	CREATE=1
+}
+
+-- Execute a program found using command PATH search, like the shell.
+-- Return the pid, as well as the I/O streams as pipes or nil on error.
+function M.popen3(policies, path, ...)
+	local pipes = {}
+	local intern = {}
+	local extern = {}
+
+	for fd, policy in pairs(policies) do
+		if M.PipePolicies.CREATE==policy then
+			pipes[fd]={posix_unistd.pipe()}
+			if posix_unistd.STDIN_FILENO==fd then
+				intern[fd]=pipes[fd][1]
+				extern[fd]=pipes[fd][2]
+			else
+				intern[fd]=pipes[fd][2]
+				extern[fd]=pipes[fd][1]
+			end
+		end
+	end
+
+	-- intern: r0, w1, w2
+	-- extern: w0, r1, r2
+
+	local pid = posix_unistd.fork()
+
+	if pid < 0 then
+		return nil
+	elseif pid == 0 then
+		local null=-1;
+		if M.contains(policies, M.PipePolicies.DISCARD) then
+			-- only open, if there's anything to discard
+			null = posix_fcntl.open('/dev/null', posix_fcntl.O_WRONLY)
```

Comment:
stdin should be supported as well; `O_WRONLY` will be incorrect in this case.

Reply:
Reading `/dev/null` to feed the result to the process?

Reply:
resolved in 63d44093162078b04492e0645179f667dfd86159